uber/tests/models/test_attendee.py

```diff
@@ -1,28 +1,43 @@
 from uber.tests import *
 
 
-def test_badge_cost(monkeypatch):
-    monkeypatch.setattr(c, 'get_oneday_price', Mock(return_value=111))
-    monkeypatch.setattr(c, 'get_attendee_price', Mock(return_value=222))
-    assert 111 == Attendee(badge_type=c.ONE_DAY_BADGE).badge_cost
-    assert 222 == Attendee().badge_cost
-    assert 333 == Attendee(overridden_price=333).badge_cost
-    assert 0 == Attendee(paid=c.NEED_NOT_PAY).badge_cost
-    assert 0 == Attendee(paid=c.PAID_BY_GROUP).badge_cost
-
-
-def test_total_cost(monkeypatch):
-    monkeypatch.setattr(c, 'get_attendee_price', Mock(return_value=10))
-    assert 10 == Attendee().total_cost
-    assert 15 == Attendee(amount_extra=5).total_cost
-
-
-def test_amount_unpaid(monkeypatch):
-    monkeypatch.setattr(Attendee, 'total_cost', 50)
-    assert 50 == Attendee().amount_unpaid
-    assert 10 == Attendee(amount_paid=40).amount_unpaid
-    assert 0 == Attendee(amount_paid=50).amount_unpaid
-    assert 0 == Attendee(amount_paid=51).amount_unpaid
```

Comment:
I merged the existing cost tests into a test class so that I could factor out some of the monkeypatching into an autouse fixture.  I then added two new test methods, one to test the case where we apply a partial discount, and one where we make the registration free (but not any extras like a tshirt).

One thing I'll note about the way I implemented this logic (and the tests reflect this) is that discount is applied even to an ``overridden_price``.  I believe this is the best and most correct approach, since otherwise the person creating the discounted registration (e.g. "register at Otakon and get $10 off") would have to remember the additional age discounts to apply, and that's a lot to expect people to remember.